components/application-broker/internal/broker/provision.go

```diff
@@ -229,7 +245,19 @@ func (svc *ProvisionService) do(iID internal.InstanceID, opID internal.Operation
 	if err := svc.enableDefaultKnativeBroker(ns); err != nil {
 		instanceState = internal.InstanceStateFailed
 		opState = internal.OperationStateFailed
-		opDesc = fmt.Sprintf("provisioning failed while enabling default Knative Broker for namespace: %s on error: %s", ns, err)
+		opDesc = fmt.Sprintf("provisioning failed while enabling default Knative Broker for namespace: %s"+
+			" on error: %s", ns, err)
+		svc.updateStates(iID, opID, instanceState, opState, opDesc)
+		return
+	}
+
+	// Create istio policy
+	if err := svc.createIstioPolicy(ns); err != nil {
+		svc.log.Printf("Error creating istio policy: %v", err)
```

Comment:
Isn't this `svc.log.Printf` redundant as `opDesc` is actually meant for that?

Reply:
@sayanh But the `opDesc` is not logged anywhere, no?
BTW, I changed it to `svc.log.Errorf("Error creating istio policy: %v", err)`

Reply:
Sure, the log doesn't give any value as the error is already logged in `updateStates`